File: Cores/DotNet/Cybernatics/RealValuedDynamicSystem.cs
using System;
using OSOL.Extremum.Cores.DotNet.Vectors;

namespace OSOL.Extremum.Cores.DotNet.Cybernatics
{
    public class RealValuedDynamicSystem: DynamicSystem<double>
    {
        public RealValuedDynamicSystem(Func<VectorObject<double>, VectorObject<double>> f, Func<VectorObject<double>, VectorObject<double>> u, ButcherTableau butcherTableau)
        {
            this.f = f;
            this.u = u;
            this.butcherTableau = butcherTableau;
        }
        
        public sealed override double BaseToDouble(double b) => b;

        public sealed override double DoubleToBase(double d) => d;
    }
}

File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Cybernatics/RealValuedDynamicSystem.scala
package OSOL.Extremum.Cores.JVM.Cybernatics

import OSOL.Extremum.Cores.JVM.Vectors.VectorObject

class RealValuedDynamicSystem(f: VectorObject[Double] => VectorObject[Double], u: VectorObject[Double] => VectorObject[Double], butcherTableau: ButcherTableau)
  extends DynamicSystem[Double](f, u, butcherTableau) {

  final override implicit def doubleToBase(d: Double): Double = d

  final override implicit def baseToDouble(b: Double): Double = b

}
